src/libcore/iter/sources.rs

```diff
@@ -57,6 +57,12 @@ unsafe impl<A: Clone> TrustedLen for Repeat<A> {}
 ///
 /// [`take`]: trait.Iterator.html#method.take
 ///
+/// If the element type of the iterator you need does not implement `Clone`,
+/// or if you do not want to keep the repeated element in memory, you can
+/// instead use the [`repeat_with`] function.
```

Comment:
Not necessary that we mention it, but another case is indeed to emit a different value each time the function is called, for example (stolen from itertools so it uses while_some() as well):

```rust
let mut heap = BinaryHeap::from(vec![2, 5, 3, 7, 8]);

// extract each element in sorted order
for element in repeat_with(|| heap.pop()).while_some() {
    print!("{}", element);
}
```

Reply:
Pretty neat example - let's have `.while_some()` in core?

Reply:
Hopefully there's a version using `Try` that can go well with Option and Result and friends.  I wonder if there's a more fundamental operation too.  Possibly-related: https://github.com/rust-lang/rust/issues/45594#issuecomment-355420120

Reply:
@scottmcm Perhaps `Try<Ok = Elt, Error: Into<()>>`?